### PR TITLE
Extra tests

### PR DESCRIPTION
**Pull request by robertprice**
Some extra tests to increase code coverage.

### CODE REVIEW

**Comment by robertprice** [comment]
This requires #1978 to be merged to fix the php7 build issues.

File: tests/Http/RequestTest.php [hunk 1]
```diff
@@ -165,14 +165,17 @@ public function testGetMethodWithOverrideParameterFromBodyObject()
     {
         $uri = Uri::createFromString('https://example.com:443/foo/bar?abc=123');
         $headers = new Headers([
-            'Content-Type' => 'application/x-www-form-urlencoded',
+            'Content-Type' => 'application/json',
```
Comment (akrabat) [inline tests/Http/RequestTest.php:168]: Why did you change this? Practically, _METHOD overrides are only going to be used with HTML forms.

Comment (robertprice) [reply tests/Http/RequestTest.php:168]: I was testing decoding of objects, but I can't remember the reasons now. 

I think it's safe for you to ignore this, and for me to learn to write better commit messages.
